fedimint-server/src/config/api.rs

```diff
@@ -303,12 +315,34 @@ impl ConfigGenApi {
             .values()
             .cloned()
             .collect();
-        if user_hashes == hashes {
-            *state = ConfigApiState::RunningConsensus(auth);
-            Ok(())
-        } else {
-            Self::bad_request("Config verification failed")
+
+        if user_hashes != hashes {
+            return Self::bad_request("Config verification failed");
+        }
+
+        let auth = config.private.api_auth.0.clone();
+        fs::write(self.data_dir.join(SALT_FILE), random_salt())
+            .map_err(|e| ApiError::server_error(format!("Unable to write to data dir {e:?}")))?;
+        write_server_config(&config, self.data_dir.clone(), &auth, &self.registry)
+            .map_err(|e| ApiError::server_error(format!("Unable to encrypt configs {e:?}")))
+    }
+
+    /// Creates the Fedimint server, decrypting the config files from disk using
+    /// the auth string.  We send a message to the `run_server` task to shutdown
+    /// the API so that the Fedimint API can start.
+    pub async fn start_consensus(&self, auth: ApiAuth) -> ApiResult<()> {
+        {
+            let cfg = read_server_config(&auth.0, self.data_dir.clone())
+                .map_err(|e| ApiError::bad_request(format!("Unable to decrypt configs {e:?}")))?;
+
+            let mut state = self.state.lock().expect("lock poisoned");
```

Comment:
So panicking is the right think to do here because once the mutex is poisoned we can't really recover?

Reply:
Yeah, I don't think there is a way to recover from them in a way that makes sense because it means I panicked in some other thread that was holding the lock.  I tried to handle the API calls in ways that don't ever panic, but return errors.